### ISSUE TITLE
`prefer-const` doesn't work on `var`s

### ISSUE THREAD

**Issue by cwgorman** [eslint/eslint#5841]
So a project I'm working on has a rather unusual dev environment, where we support `var` and `const`, but not `let`. The reason being that [`const` is supported](http://caniuse.com/#search=const) (at least partially) on all the browsers we support, but [`let` is not](http://caniuse.com/#search=let).

So we'd like to use `const` when possible, and the `prefer-const` rule seems to be exactly what we need, except that it doesn't seem to do anything to `var`s.

**What version of ESLint are you using?**
2.7.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```yaml
parserOptions:
  ecmaVersion: 6
rules:
  prefer-const: 2
```

**What did you do? Please include the actual source code causing the issue.**
```js
'use strict';

var thing = 7;
thing = 2;
console.log(thing);
//Should catch that this can be const
var otherThing = 5;
console.log(otherThing);
```

**What did you expect to happen?**
I would expect ESLint to catch that `otherThing` can be declared as `const` instead of `var`.

**What actually happened? Please include the actual, raw output from ESLint.**
It just passes.

**Comment by nzakas** [comment]
This is by design. You can't just replace `var` with `const` because they have different rules around scoping. Specifically, `const` is block-scoped while `var` is function-scoped.

**Comment by cwgorman** [comment]
@nzakas Could the rule not detect that the `var` is 1) never reassigned, and 2) used as block-scoped? I'd imagine it can, since the rule `block-scoped-var` exists. Or perhaps if `block-scoped-var` is also enabled, do this?
